internal/vds/vds_test.go

```diff
@@ -427,81 +427,55 @@ func TestInvalidFence(t *testing.T) {
-		expected []float32
-		method   string
-	}{
-		{
-			expected: []float32{
-				100, 101, 102, 103,
-				108, 109, 110, 111,
-				116, 117, 118, 119,
-				0, 0, 0, 0,
-			},
-			method: "nearest",
-		},
-		{
-			expected: []float32{
-				100, 101, 102, 103,
-				106, 107, 108, 109,
-				116, 117, 118, 119,
-				0, 0, 0, 0,
-			},
-			method: "linear",
-		},
-		{
-			expected: []float32{
-				99.338, 100.338, 101.338, 102.338,
-				105.448, 106.448, 107.448, 108.448,
-				115.706, 116.706, 117.706, 118.706,
-				0, 0, 0, 0,
-			},
-			method: "cubic",
-		},
-		{
-			expected: []float32{
-				114, 108, 109, 110,
-				114.54, 108.54, 109.54, 110.54,
-				109, 110, 111, 112,
-				0, 0, 0, 0,
-			},
-			method: "angular",
-		},
-		{
-			expected: []float32{
-				100, 101, 102, 103,
-				109.2, 110.2, 111.2, 112.2,
-				116, 117, 118, 119,
-				0, 0, 0, 0,
-			},
-			method: "triangular",
-		},
-	}
-
-	for _, testCase := range testCases {
-		interpolationMethod, _ := GetInterpolationMethod(testCase.method)
-		buf, err := Fence(well_known, "ij", fence, interpolationMethod)
-		if err != nil {
-			msg := "[%v]Unable to retrieve fence %v"
-			t.Errorf(msg, testCase.method, fence)
-		}
-		interpolated, err := toFloat32(buf)
-		if err != nil {
-			msg := "[%v]Err: %v"
-			t.Errorf(msg, testCase.method, err)
-		}
-		if len(testCase.expected) != len(*interpolated) {
-			msg := "[%v]Interpolated does not match length of expected"
-			t.Errorf(msg, testCase.method)
-		}
-		for i, v := range *interpolated {
-			if v != testCase.expected[i] {
-				msg := "[%v]Interpolated does not match expected"
-				t.Errorf(msg, testCase.method)
+func TestFenceInterpolation(t *testing.T) {
+	fence := [][]float32{{3.2, 3}, {3.2, 6.3}, {1, 3}, {3.2, 3}, {5.4, 3}}
+	interpolationMethods := []string{"nearest", "linear", "cubic", "triangular", "angular"}
+	for i, v1 := range interpolationMethods {
+		interpolationMethod, _ := GetInterpolationMethod(v1)
+		buf1, _ := Fence(well_known, "cdp", fence, interpolationMethod)
+		for j, v2 := range interpolationMethods {
+			if i <= j {
+				continue
+			}
+			interpolationMethod, _ := GetInterpolationMethod(v2)
+			buf2, _ := Fence(well_known, "ij", fence, interpolationMethod)
+			different := false
+			for k, _ := range buf1 {
+				if buf1[k] != buf2[k] {
+					different = true
+				}
+			}
+			if !different {
+				msg := "[fence_interpolation]Expected %v interpolation and %v interpolation to be different"
+				t.Errorf(msg, v1, v2)
 			}
 		}
 	}
+}
+
+func TestFenceInterpolationDefaultIsNearest(t *testing.T) {
+	defaultInterpolation, _ := GetInterpolationMethod("")
+	nearestInterpolation, _ := GetInterpolationMethod("nearest")
+	if defaultInterpolation != nearestInterpolation {
+		msg := "[fence_interpolation]Default interpolation is not nearest"
+		t.Error(msg)
+	}
+}
 
+func TestInvalidInterpolationMethod(t *testing.T) {
+	interpolation := "sand"
+	_, err := GetInterpolationMethod(interpolation)
+	if err == nil {
```

Comment:
Small comment, feel free to disregard if you don't like it, but I would vote for additionally testing important part of error message, like:
```
if !strings.Contains(err.Error(), "Invalid interpolation method")
  complain message is wrong
```

Whether to test or not error messages is often a personal preference. I by default prefer to do so, but many do not.

Reply:
I added this as i can see it has been done in other tests